catalog/transformer/catalog-transformer-html/src/main/java/ddf/catalog/transformer/html/HtmlMetacardUtility.java

```diff
@@ -0,0 +1,177 @@
+import com.github.jknack.handlebars.ValueResolver;
+import com.github.jknack.handlebars.context.FieldValueResolver;
+import com.github.jknack.handlebars.context.JavaBeanValueResolver;
+import com.github.jknack.handlebars.context.MapValueResolver;
+import com.github.jknack.handlebars.helper.IfHelper;
+import com.github.jknack.handlebars.io.ClassPathTemplateLoader;
+import com.github.jknack.handlebars.io.TemplateLoader;
+import ddf.catalog.transformer.html.models.HtmlBasicValueModel;
+import ddf.catalog.transformer.html.models.HtmlEmptyValueModel;
+import ddf.catalog.transformer.html.models.HtmlExportCategory;
+import ddf.catalog.transformer.html.models.HtmlMediaModel;
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Utility class that loads templates and renders html based on object models. */
+public class HtmlMetacardUtility {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(HtmlMetacardUtility.class);
+
+  private TemplateLoader templateLoader;
+
+  private Handlebars handlebars;
+
+  private ValueResolver[] resolvers;
+
+  private Template template;
+
+  private static MimeType mimeType;
+
+  private String htmlTemplate;
+
+  private static final String TEMPLATE_DIRECTORY = "/templates";
+
+  private static final String TEMPLATE_SUFFIX = ".hbs";
+
+  private static final String HTML_TEMPLATE = "template";
+
+  private List<HtmlExportCategory> categoryList;
+
+  static {
+    try {
+      mimeType = new MimeType("text/html");
+    } catch (MimeTypeParseException e) {
+      LOGGER.warn("Failed to apply mimetype text/html");
+    }
+  }
+
+  public HtmlMetacardUtility() {
+    this(HTML_TEMPLATE);
+  }
+
+  public HtmlMetacardUtility(List<HtmlExportCategory> categoryList) {
+    this(HTML_TEMPLATE);
+    this.categoryList = categoryList;
+  }
+
+  public HtmlMetacardUtility(String htmlTemplate) {
+    this.htmlTemplate = htmlTemplate;
+
+    this.templateLoader = new ClassPathTemplateLoader();
+    this.templateLoader.setPrefix(TEMPLATE_DIRECTORY);
+    this.templateLoader.setSuffix(TEMPLATE_SUFFIX);
+
+    this.handlebars = new Handlebars(this.templateLoader);
+
+    this.resolvers =
+        new ValueResolver[] {
+          FieldValueResolver.INSTANCE, MapValueResolver.INSTANCE, JavaBeanValueResolver.INSTANCE
+        };
+
+    this.registerHelpers();
+
+    try {
+      this.template = this.handlebars.compile(htmlTemplate);
+    } catch (IOException e) {
+      LOGGER.warn("Failed to compile handlebars template {}", HTML_TEMPLATE, e);
+    }
+  }
+
+  public static List<HtmlExportCategory> sortCategoryList(List<HtmlExportCategory> categories) {
+    if (categories != null) {
+      return categories
+          .stream()
+          .sorted(Comparator.comparing(HtmlExportCategory::getTitle))
```

Comment:
❓ Do you need a new list at the end of this process or can you modify the input list to sort it? If the latter, could just call `categories.sort(Comparator.comparing(HtmlExportCategory::getTitle))`

Reply:
I thought it might be best to let the transformer decide if it wanted a list of categories sorted alphabetically by title. Also doing it this way addressed one of @Bdthomson comments